crates/ruff/src/rules/flake8_comprehensions/rules/unnecessary_map.rs

```diff
@@ -103,11 +103,22 @@ pub(crate) fn unnecessary_map(
                 return;
             };
 
-            if parameters
-                .as_ref()
-                .is_some_and(|parameters| late_binding(parameters, body))
```

Comment:
Nit: Let's revert to the `is_some_and` pattern, I think we intentionally migrated towards that style?